### PR TITLE
AppRootPage: Fix passing the queryParams

### PR DESCRIPTION
**Pull request by leventebalogh**
### What is the problem?

While [updating the routing for app plugins](https://github.com/grafana/grafana/pull/57771/files#diff-1cf7e00838f17a49eba2776a92eab85653b4894158fad3e4ba4f3245bc14af8eR38) I have introduced a bug: accidentally passing down the route params instead of the key-value pairs of query parameters. This caused plugins relying on the `queryParams` prop not work. ([Example issue](https://github.com/grafana/gex-plugins/issues/844))

### What changed?

- Pushing the correct query parameters down
- Only re-create the query params object if the `location.search` changes

### CODE REVIEW

**Review by darrenjaneczek — APPROVED** [review]
Thank you!

File: public/app/features/plugins/components/AppRootPage.tsx [hunk 1]
```diff
@@ -157,4 +157,14 @@ export function getAppPluginPageError(meta: AppPluginMeta) {
   return null;
 }
 
+function convertLocationSearchToObject(locationSearch: string) {
```
Comment (jackw) [inline public/app/features/plugins/components/AppRootPage.tsx:160]: nit: I think we have a utility function available in runtime [here](https://github.com/grafana/grafana/blob/main/packages/grafana-runtime/src/services/LocationService.ts#L125-L140) that does a similar job. However this feels easier to reason with so I'm wondering if in the future we should replace the util in runtime with this implementation. WDYT?

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: public/app/features/plugins/components/AppRootPage.tsx (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/plugins/components/AppRootPage.tsx:160]: Sounds like a good suggestion bro! 👍

**Pull request merged**